app/models/piece.rb

```diff
@@ -6,10 +6,23 @@ def self.types
     %w[King Queen Rook Bishop Knight Pawn]
   end
 
-  def self.colors
+  def self.color
     %w[White Black]
   end
 
+  def move_to!(piece, destination_x, destination_y)
+    destination_piece_check = Piece.find_by(column_coordinate: destination_x, row_coordinate: destination_y).present?
```

Comment:
`Piece.find_by` will look at all pieces in our database.

We want to only check pieces that are a part of the current game.

`game.pieces.find_by(...)`